src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetUserProfileOperation.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.ccd.domain.service.aggregated;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.aggregated.UserProfile;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.service.common.AccessControlService;
+
+@Service
+@Qualifier(AuthorisedGetUserProfileOperation.QUALIFIER)
+public class AuthorisedGetUserProfileOperation implements GetUserProfileOperation {
+    public static final String QUALIFIER = "authorised";
+
+    private final UserRepository userRepository;
+    private final AccessControlService accessControlService;
+    private final GetUserProfileOperation getUserProfileOperation;
+
+    public AuthorisedGetUserProfileOperation(@Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository,
+                                             AccessControlService accessControlService,
+                                             @Qualifier(DefaultGetUserProfileOperation.QUALIFIER) GetUserProfileOperation getUserProfileOperation) {
+        this.accessControlService = accessControlService;
+        this.getUserProfileOperation = getUserProfileOperation;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserProfile execute(final String userToken) {
+        return filterCaseTypes(getUserProfileOperation.execute(userToken));
+    }
+
+    private UserProfile filterCaseTypes(UserProfile userProfile) {
+        final Set<String> userRoles = getUserRoles();
+        Arrays.stream(userProfile.getJurisdictions()).forEach(
+            jurisdictions -> jurisdictions.setCaseTypes(
+                jurisdictions.getCaseTypes()
+                    .stream()
+                    .map(caseType -> verifyAccess(caseType, userRoles, CAN_READ))
```

Comment:
Here we are testing if the user `CAN_READ`, however, this is not relevant if the case types or events are used for creating cases as you could have READ write on case type or event, but be missing the Create right.

Reply:
updated the Access Type to CAN_CREATE